torch/csrc/jit/python/python_sugared_value.cpp

```diff
@@ -235,6 +235,30 @@ SugaredValuePtr ModuleValue::getitem(
     Value* idx) {
   if (concreteType_->getIterableModuleKind() == IterableModuleKind::LIST) {
     return getSugaredDict(loc, m)->getModules()->getitem(loc, m, idx);
+  } else if (
+      concreteType_->getIterableModuleKind() == IterableModuleKind::DICT) {
+    if (idx->type()->kind() == c10::TypeKind::StringType) {
```

Comment:
yup. actually, i think i can totally drop the StringType check, as toIValue does a type() == StringType inside, and will return none otherwise.  see latest rev once i push.